### ISSUE TITLE
How to filter via the pivotwhere filter?

### ISSUE THREAD

**Issue by stefan-beech** [laravel-json-api/laravel#119]
I'm trying to filter my record Company via a belongstomany relation with the record Sector, Which guided me to the WherePivot filter. The documentation around this part of the filters isn't exactly what I call very broad and I've looked into the the dummy project and found a relation which uses the belongstomany method. The problem there is that there isn't an example how filtering is done through this relation. so first I want to know what the correct syntaxes is for calling a filter via a pivot.
something like ? `localhost/api/v1/companies?filter[sectors][name]=test` ? secondly If I try to access this url I get an message back regarding sectors not being a valid filter.

`{
"jsonapi": {
"version": "1.0"
},
"errors": [
{
"detail": "Filter parameter sectors is not allowed.",
"source": {
"parameter": "filter"
},
"status": "400",
"title": "Invalid Query Parameter"
}
]
}`

I will provide some extra context here

Company schema fields:
<pre>
`public function fields(): array
    {
        return [
            ID::make(),
            Str::make('name'),
            Boolean::make('template'),
            Number::make('created_by')->readOnly(),
            DateTime::make('created_at')->sortable()->readOnly(),
            Number::make('updated_by')->readOnly(),
            DateTime::make('updated_at')->sortable()->readOnly(),
            Number::make('deleted_by')->readOnly(),
            DateTime::make('deleted_at')->sortable()->readOnly(),
            HasMany::make('addresses'),
            BelongsToMany::make('sectors')->withFilters(
                WherePivot::make('name')
            ),
            BelongsToMany::make('products'),
        ];
    }`
</pre>

Company request rules:
<pre>
`public function rules(): array
    {
        return [
            'name' => ['required', 'string'],
            'template' => ['boolean'],
            'addresses' => ['nullable', 'array'],
            'sectors' => JsonApiRule::toMany(),
            'products' => ['nullable', 'array'],
        ];
    }`
</pre>

**Comment by lindyhopchris** [comment]
Pivot filters are applied to the relationship, so you'd use it on the relationship endpoints:

```
/api/v1/companies/1/sectors?filter[name]=test
/api/v1/companies/1/relationships/sectors?filter[name]=test
```

You're right that this could be clearer in the docs, so I'll leave this issue open as a reminder to add to the docs.

If you want to use a filter in the way that you're trying, then you should look at the `WhereHas` filter.

**Comment by lindyhopchris** [comment]
I think this is most likely an issue related to #139 which is now resolved - will be tagged as `v1.0.1`. If the problem persists after upgrading, please reopen the issue.
